### PR TITLE
glfw3: migrate from homebrew/versions.

### PR DESCRIPTION
**Pull request by JCount**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Also, rename to `glfw`.

Fixes https://github.com/Homebrew/homebrew-versions/issues/1318.

### CODE REVIEW

File: Formula/glfw.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  sha256 "e10f0de1384d75e6fc210c53e91843f6110d6c4f3afbfb588130713c2f9d8fe8"
+
+  head "https://github.com/glfw/glfw.git"
+
+  bottle do
+    cellar :any
+    sha256 "299adbffdf82bc6040b126d317cd90cd6569fe0e4e16ce64fdc06e926797f805" => :el_capitan
+    sha256 "e1634c7ad4cde63222d193f91124066575412a2cc4889cccc8768dd977e5acfd" => :yosemite
+    sha256 "a276c4098579c17da6105c0d5f9372a5dc7c08189b6fe05b366e20530c582511" => :mavericks
+  end
```
Comment (MikeMcQuaid) [inline Formula/glfw.rb]: Can remove this as it's effectively a new formula now.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
You can ignore this audit failure: https://bot.brew.sh/job/Homebrew%20Core/12523/version=any_sierra/testReport/junit/brew-test-bot/sierra/audit_glfw___new_formula/

One nit and then we can ship this.

File: Formula/glfw.rb (hunk 1 above)
Comment (JCount) [inline Formula/glfw.rb]: Ok

**Comment by JCount** [comment]
Should be good to go.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by DomT4** [comment]
This could have probably used some `link_overwrite`s:

```
==> Finishing up
rm /usr/local/include/GLFW
ln -s ../../Cellar/glfw3/3.2.1/include/GLFW/glfw3.h glfw3.h
ln -s ../../Cellar/glfw3/3.2.1/include/GLFW/glfw3native.h glfw3native.h
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink include/GLFW/glfw3.h
Target /usr/local/include/GLFW/glfw3.h
is a symlink belonging to glfw3. You can unlink it:
  brew unlink glfw3

To force the link and overwrite all conflicting files:
  brew link --overwrite glfw

To list all files that would be deleted:
  brew link --overwrite --dry-run glfw

Possible conflicting files are:
/usr/local/include/GLFW/glfw3.h -> /usr/local/Cellar/glfw3/3.2.1/include/GLFW/glfw3.h
/usr/local/include/GLFW/glfw3native.h -> /usr/local/Cellar/glfw3/3.2.1/include/GLFW/glfw3native.h
/usr/local/lib/cmake/glfw3/glfw3Config.cmake
/usr/local/lib/cmake/glfw3/glfw3ConfigVersion.cmake
/usr/local/lib/cmake/glfw3/glfw3Targets-release.cmake
/usr/local/lib/cmake/glfw3/glfw3Targets.cmake
/usr/local/lib/cmake/glfw3/glfw3Config.cmake
/usr/local/lib/cmake/glfw3/glfw3ConfigVersion.cmake
/usr/local/lib/cmake/glfw3/glfw3Targets-release.cmake
/usr/local/lib/cmake/glfw3/glfw3Targets.cmake
/usr/local/lib/pkgconfig/glfw3.pc -> /usr/local/Cellar/glfw3/3.2.1/lib/pkgconfig/glfw3.pc
==> Summary
🍺  /usr/local/Cellar/glfw/3.2.1: 14 files, 286.5K
```

**Comment by JCount** [comment]
@DomT4 for future reference, how would that be done?

**Comment by ilovezfs** [comment]
That looks more like a brew bug, not a formula bug.

**Comment by DomT4** [comment]
It could probably be solved from the brew side, yes, but Homebrew has used/abused `link_overwrite` before and it'd have probably worked here too. Just wanted to flag it up anyway because it's a pain to spot, especially when 4/5 things get updated in a row.

**Comment by ilovezfs** [comment]
I can't wait to see how many issues this causes to be filed.

**Comment by DomT4** [comment]
I didn't even notice it until I checked why zsh hadn't proceeded onto the second part of my `brewup && dirtypub` command, but I do upgrade verbosely so perhaps less masked for the regular use case.

**Comment by ilovezfs** [comment]
>dirtypub

:laughing:

**Comment by DomT4** [comment]
Is an alias for `brew cleanup --prune=1 && brew prune` 😸

**Pull request closed without merging**